### PR TITLE
[Owners] Rename Digital Pattern folders everywhere

### PR DESCRIPTION
**Pull request by ni-pkhare**
### What does this Pull Request accomplish?

Rename the digital pattern folder to `nidigitalpattern`
Files/folders that are changed :
1) Inside **example folder** : folder for digital pattern is renamed
2) Inside **Generated folder** : folder for digital pattern is renamed, Also all the generated files are now prefixed with **nidigitalpattern**  ex `nidigital_service.cpp` changed to `nidigitalpattern_service.cpp` , similarly for all other files.
3) Inside **souce/codegen/metadata folder** : folder for digital pattern metadata is renamed.
4) Inside **source/custom folder** : `nidigital_service.custom.cpp` is renamed to `nidigitalpattern_service.custom.cpp`

Note : `nidigital_driver_api_tests.cpp` , `nidigital_session_tests.cpp` and `core_server.cpp` includes generated header files, since names of those are changed now so I have updated these files with the appropriate names. 

### Why should this Pull Request be merged?

As per discussion with PO, `nidigitalpattern` looks more clear and understandable when compared to `nidigital`

### What testing has been done?
Manual

### CODE REVIEW

**Review by doshirohan — CHANGES_REQUESTED** [review]
You need to update metadata/codegen so that files get generated in new location.

And looking at build failure in PR, also update CMakeLists.txt so that digitalpattern files are picked up from new generated location.
> FileNotFoundError: [Errno 2] No such file or directory: 'D:/a/grpc-device/grpc-device/source/codegen/metadata/nidigital/__init__.py'

**Review by kruthi-NI — APPROVED** [review]

**Comment by ni-pkhare** [comment]
> You need to update metadata/codegen so that files get generated in new location.
> 
> And looking at build failure in PR, also update CMakeLists.txt so that digitalpattern files are picked up from new generated location.
> 
> > FileNotFoundError: [Errno 2] No such file or directory: 'D:/a/grpc-device/grpc-device/source/codegen/metadata/nidigital/**init**.py'

@doshirohan , I have made changes at appropriate places. The build is also successful. Could you please have a look at it?

File: source/codegen/metadata/nidigitalpattern/config.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
         }
     },
     'metadata_version': '2.0',
-    'module_name': 'nidigital',
+    'module_name': 'nidigitalpattern',
```
Comment (doshirohan) [inline source/codegen/metadata/nidigitalpattern/config.py:46]: We should record this change in CHANGES.md

**Review by doshirohan — APPROVED** [review]

File: generated/nidigitalpattern/nidigitalpattern_library.cpp [hunk 2]
```diff
@@ -3,7 +3,7 @@
 //---------------------------------------------------------------------
 // Service implementation for the NI-Digital Pattern Driver Metadata
 //---------------------------------------------------------------------
-#include "nidigital_library.h"
+#include "nidigitalpattern_library.h"
 
 #if defined(_MSC_VER)
 static const char* kLibraryName = "niDigital_64.dll";
```
Comment (doshirohan) [inline generated/nidigitalpattern/nidigitalpattern_library.cpp:9]: @jonlugoatx should namespace on line 14 below be updated to `nidigitalpattern_grpc` as well?

**Review by reckenro — APPROVED** [review]

**Comment by reckenro** [comment]
FYI, Charlie (@SparkingSpork ) is out of the office for the next few days so you don't need to wait for him after all other code owners aprove.

**Review by kyledsullivan — APPROVED** [review]

File: generated/nidigitalpattern/nidigitalpattern_library.cpp (hunk 2 above)
Comment (jonlugoatx) [reply generated/nidigitalpattern/nidigitalpattern_library.cpp:9]: Yes, we should keep the module name and namespace consistent

**Review by jonlugoatx — APPROVED** [review]

File: generated/nidigitalpattern/nidigitalpattern_library.cpp (hunk 2 above)
Comment (ni-pkhare) [reply generated/nidigitalpattern/nidigitalpattern_library.cpp:9]: Sure. I have changed the namespace name and made changes at appropriate location.

File: source/codegen/metadata/nidigitalpattern/config.py (hunk 1 above)
Comment (ni-pkhare) [reply source/codegen/metadata/nidigitalpattern/config.py:46]: Done. I forgot to do it earlier.

**Pull request merged**